include/cuco/detail/open_addressing/open_addressing_impl.cuh

```diff
@@ -445,9 +445,24 @@ class open_addressing_impl {
                       Ref container_ref,
                       cuda_stream_ref stream) const noexcept
   {
+    auto const num_keys = cuco::detail::distance(first, last);
+    if (num_keys == 0) { return; }
+
+    auto const grid_size   = cuco::detail::grid_size(num_keys, cg_size, 1, 1024);
     auto const always_true = thrust::constant_iterator<bool>{true};
-    this->contains_if_async(
-      first, last, always_true, thrust::identity{}, output_begin, container_ref, stream);
+
+    auto const bytes =
+      sizeof(typename storage_ref_type::window_type) * this->storage_.num_windows();
+
+    if (bytes >= 48000) {
+      open_addressing_ns::detail::contains_if_n<false, cg_size, cuco::detail::default_block_size()>
+        <<<grid_size, cuco::detail::default_block_size(), 0, stream>>>(
```

Comment:
Here we specify `cuco::detail::default_block_size()` which is 128 but earlier we used 1024 to calculate the grid size.